aspnetcore/fundamentals/dependency-injection.md

```diff
@@ -220,6 +220,38 @@ In the following example, the call to `AddSingleton` registers `MyDependency` as
 services.AddSingleton<IMyDependency, MyDependency>();
```

Comment:
This doc should explain what happens when you register multiple services for an interface before it tries to describe how the TryAdd* overloads are different.

E.g.:
- Add{Lifetime} registers a service
- Calling Add{Lifetime} again registers another instance of that service. The last one registered will be used in most situations.
- Resolving `IEnumerable<{MyService}>` will return all services registered for that type in the order they were registered.
- TryAdd{Lifetime} avoids adding a service if there is already something registered for that interface
- TryAddEnumerable...

Reply:
I was thinking the same thing. I went for the smaller change initially, but I'll try reordering and see if it reads better.

Reply:
The `Multiple implementations` column above isn't explicitly explained and could be worked into this description.

Reply:
I don't entirely understand the "Multiple implementations" column. Anything other than the TryAdd methods can add multiple implementations of a service. It would be a bit weird to do so if you aren't specifying different service and implementation types, but nothing's stopping you.

Reply:
I guess I'm probably confusing "implementations" and "instances".

Reply:
Do we add a "Multiple instances" column that's distinct from the "Multiple implementations" column? I feel that just confuses things further. I don't really want to touch that with this PR b/c then we start getting into rewrite territory.

Reply:
Filing follow-up issues works too

Reply:
I fixed the ordering and added a sentence about the distinction between "implementations" and instances." I'm not sure exactly how you want done with the table. Can you please file the issue if you still think we need one?